libs/playlist.liq

```diff
@@ -100,45 +109,47 @@ def playlist.reloadable(~id="",~random=false,~on_done={()},
 # @param ~conservative If true, estimated remaining time on the current track is not considered when computing queue length.
-# @param ~random Randomize playlist content
-# @param ~on_done Function to execute when the playlist is finished
-# @param ~reload_mode If set to "watch", will be reloaded when the playlist is changed
-# @param ~filter Filter out some files depending on metadata
-# @param uri Playlist URI
-def playlist.once(~id="",~random=false,~on_done={()},~reload_mode="",
-                  ~default_duration=30.,~length=10.,~conservative=false,
-                  ~filter=fun(_)->true,uri)
-  let (reload, s) = playlist.reloadable(id=id,random=random,on_done=on_done,
-                           conservative=conservative,default_duration=default_duration,
-                           filter=filter,uri)
-  if reload_mode == "watch" then
-    unwatch = file.watch(uri,fun () -> reload())
-    source.on_shutdown(s,unwatch)
-  end
-  s
-end
-
-# Play the whole playlist as one track.
-# @category Source / Track Processing
-# @param ~id Force the value of the source ID.
 # @param ~default_duration When unknown, assume this duration (in sec.) for files.
 # @param ~length How much audio (in sec.) should be queued in advance.
-# @param ~conservative If true, estimated remaining time on the current track is not considered when computing queue length.
-# @param ~random Randomize playlist content
+# @param ~mime_type Default MIME type for the playlist. Empty string means automatic detection.
+# @param ~mode Play the files in the playlist either in the order ("normal" mode), or shuffle the playlist each time it is loaded, and play it in this order for a whole round ("randomize" mode), or pick a random file in the playlist each time ("random" mode).
+# @param ~prefix Add a constant prefix to all requests. Useful for passing extra information using annotate, or for resolution through a particular protocol, such as replaygain.
+# @param ~reload Amount of time (in seconds or rounds), when applicable, before which the playlist is reloaded; 0 means never.
+# @param ~reload_mode Unit of the reload parameter, either "once" (play the playlist only once), "rounds", "seconds" or "watch" (reload the file whenever it is changed).
+# @param ~timeout Timeout (in sec.) for a single download.
 # @param uri Playlist URI.
-def playlist.merge(~id="",~random=false,
-                   ~default_duration=30.,~length=10.,~conservative=false,uri) =
-  pl = playlist.reloadable(id=id,random=random,
-                           conservative=conservative,default_duration=default_duration,
-                           uri)
-  let (reload, s) = pl
-  s = merge_tracks(s)
-  on_end(delay=0.,fun(_,_)->reload(),s)
+def playlist(~id="", ~check_next=fun(_)->true, ~conservative=false, ~default_duration=30., ~length=10., ~mime_type="", ~mode="randomize", ~prefix="", ~reload=0, ~reload_mode="seconds", ~timeout=20., uri)
```

Comment:
Let's merge `playlist.reloadable` and `playlist` when we add the new commands!

Reply:
Yeah, that was my idea, but let's take it slowly.

Reply:
👍